docs/topics/db/aggregation.txt

```diff
@@ -643,6 +643,61 @@ aggregate that author count, referencing the annotation field:
     >>> Book.objects.annotate(num_authors=Count("authors")).aggregate(Avg("num_authors"))
     {'num_authors__avg': 1.66}
 
+.. _aggregation-mysql-only-full-group-by:
+
+Aggregating with MySQL ONLY_FULL_GROUP_BY enabled
+-------------------------------------------------
+
+When you are generating an aggregate by using ``.values()`` clause to create the
+grouping for the annotation on MySQL, and the ``ONLY_FULL_GROUP_BY`` SQL mode
+is enabled, it may be necessary to apply :class:`~django.db.models.AnyValue` if
+the annotation includes a mix of aggregate and non-aggregate expressions.
+
+Take the following example:
+
+.. code-block:: pycon
+
+    >>> from django.db.models import F, Count, Greatest
+    >>> Book.objects.values(greatest_pages=Greatest("pages", 600)).annotate(
+    ...     num_authors=Count("authors"),
+    ...     pages_per_author=F("greatest_pages") / F("num_authors"),
+    ... ).aggregate(Avg("pages_per_author"))
+
+This creates groups of books based on the SQL column ``GREATEST(pages, 600)``.
+One unique group consists of books with 600 pages or less, and other unique
+groups will consist of books with the same pages. The ``pages_per_author``
+annotation is composed of aggregate and non-aggregate expressions,
+``num_authors`` is an aggregate expression while ``greatest_page`` isn't.
+
+The column in the group by clause is the expression for ``greatest_pages``.
+MySQL is not able to determine that the ``greatest_pages`` referenced in
+``pages_per_author`` is functionally dependent on the column in the group by
+clause, and it might raise a
+``OperationalError: incompatible with sql_mode=only_full_group_by`` error.
+To avoid the ``OperationalError``, you can wrap the non-aggregate expression
+with :class:`~django.db.models.AnyValue`.
```

Comment:
```suggestion
Since the grouping is based on the ``greatest_pages`` expression, MySQL may be
unable to determine that ``greatest_pages`` (used in the ``pages_per_author``
expression) is functionally dependent on the grouped column. As a result, it
may raise an error like:

.. code-block:: pytb

    OperationalError: incompatible with sql_mode=only_full_group_by

To avoid this, you can wrap the non-aggregate expression with
:class:`~django.db.models.AnyValue`.
```